src/torrent/peer/mod.rs

```diff
@@ -123,12 +123,29 @@ impl PeerConn {
     /// Creates a new "outgoing" peer, which acts as a client.
     /// Once created, set_torrent should be called.
     pub fn new_outgoing(ip: &SocketAddr) -> io::Result<PeerConn> {
+        if let Some((_, 1)) = IP_FILTER.longest_match(ip.ip()) {
```

Comment:
We should probably make the 1 a constant so it’s more clear whats matching here.